File: src/elements/root/simple-methods.ts
import { ElementTag } from "../element-tag";

export class SimpleMethods extends ElementTag {
    public static readonly TAG = "simple-methods";

    public convert(): string[] {
        return [
            `public class ${this.converter.getClassName()} {`,
            ...this.convertChildren().map(this.prependIndentationMapper),
            "}",
        ];
    }
}

File: src/main/java/co/uk/stefanpuia/minilang2java/tag/root/method/SimpleMethods.java
package co.uk.stefanpuia.minilang2java.tag.root.method;

import co.uk.stefanpuia.minilang2java.core.TagInit;
import co.uk.stefanpuia.minilang2java.core.model.MinilangTag;
import co.uk.stefanpuia.minilang2java.tag.Tag;
import java.util.List;

/** "simple-methods" is optional, this works as a wrapper only */
@MinilangTag("simple-methods")
public class SimpleMethods extends Tag {

  public SimpleMethods(final TagInit tagInit) {
    super(tagInit);
  }

  @Override
  public List<String> convertSelf() {
    return convertChildren();
  }
}
